[JavaScript] Add arrays-basic concept exercise (#1033)
exercism/v3: languages/JavaScript/exercises/concept/arrays-basic/.docs/hints.md

### 1. Retrieve a card from a stack

> - Note that the array index starts from `0`.

Reviewer: Maybe link to documentation that shows how to access items by index?

### 2. Exchange a card in the stack

> - The array is a mutable structure, you can change its content anytime.

Reviewer: Maybe it is possible to link to documentation that shows how to do this?

Author: Maybe 'Access the Elements of an Array' section from [this](https://www.w3schools.com/js/js_arrays.asp) w3schools page, but I can't find an url which scrolls into that section (no id there).

Reviewer: That resource does look nice. The other information in that document is nice too, so maybe still use it and just note that the information is in the "Changing an Array Element" section?

Reviewer 2: Perhaps https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array#Common_operations

### 3. Create a stack of cards

> - The `push` method adds a new value at the end of the array. *e.g.* `array.push(newValue)`

Reviewer: For our hints we'd like them to not give away the solution literally, but instead prefer to link to an external resource. Here is an [example](https://github.com/exercism/v3/blob/master/languages/csharp/exercises/concept/strings/.docs/hints.md) what that could look like. This same comments also applies to the three tasks following this task.

### 1. Retrieve a card from a stack

> - Note that the array index starts from `0`. [Here][access_array_elements_resource] you can find more information about how to access array elements.

Reviewer: Maybe split them into two hints?

```suggestion
- Array indices start at `0`. 
- [This page][access_array_elements_resource] has more information on how to access array elements.
```

### 2. Exchange a card in the stack

- The array is a mutable structure, you can change its content anytime.

### 3. Create a stack of cards

> - The `push` method adds a new value at the end of the array. [Here][push_method_docs] you can find documentation and an example for the `push` method.

Reviewer: It would be really nice not to give away the actual solution here. Maybe do:

```suggestion
- There is a [built-in][push_method_docs] method to add a new value to the end of the array.
```

So the reason we don't want to explain the solution directly and instead refer to external documentation, is that we then introduce the student to the fact that there is external documentation and that becomes a starting point for the student to know where to look for when wanting to try and see what other methods there are.

Author: This approach makes sense!

 
### 4. Remove a card from the stack

- `splice` is a versatile method, which among other use cases, can be used to remove elements starting at a certain position. [Here][splice_method_docs] you can find documentation and an example for the `splice` method.

### 5. Remove the top card from the stack

> - The `pop` method removes the last element from the array. [Here][pop_method_docs] you can find documentation and an example for the `pop` method.

Reviewer: ```suggestion
- There is a [built-in][push_method_docs] method to remove the last element from the array.
```

### 6. Check size of the stack

> - Any array's length can be determined by using the `length` property on the array object. [Here][length_property_docs] you can find documentation and an example for the `length` property.

Reviewer: ```suggestion
- Arrays have a [property][length_property_docs] to retrieve their length.
```

### 1. Retrieve a card from a stack

- Array indices start at `0`. 
- [This page][access_array_elements_resource] has more information on how to access array elements.

### 2. Exchange a card in the stack

- The array is a mutable structure, you can change its content anytime. 
- You can find an example [here][change_array_elements_resource], inside the 'Changing an Array Element' section.

### 3. Create a stack of cards

- There is a [built-in][push_method_docs] method to add a new value to the end of the array.
 
### 4. Remove a card from the stack

> - `splice` is a [built-in][splice_method_docs] versatile method, which among other use cases, can be used to remove elements starting at a certain position.

Reviewer: ```suggestion
- There is a [built-in][splice_method_docs] method that, among other use cases, can be used to remove elements starting at a certain position.
```
